# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentLoadListener.h
# Commit: 1177913e1edf
# Full Hash: 1177913e1edf443da73d438f95a334e466b623e2
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-02 21:26:33
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 8: Add list of open channels from OngoingEarlyHints to RedirectToRealChannelArgs r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161179
# ==============================================================================

diff -r d33ccbbf407d -r 1177913e1edf netwerk/ipc/DocumentLoadListener.h
--- a/netwerk/ipc/DocumentLoadListener.h	Fri Dec 02 09:45:26 2022 +0000
+++ b/netwerk/ipc/DocumentLoadListener.h	Fri Dec 02 09:45:27 2022 +0000
@@ -110,6 +110,7 @@
         mStreamFilterEndpoints;
     uint32_t mRedirectFlags;
     uint32_t mLoadFlags;
+    nsTArray<EarlyHintConnectArgs> mEarlyHints;
     RefPtr<PDocumentChannelParent::RedirectToRealChannelPromise::Private>
         mPromise;
   };
@@ -278,10 +279,10 @@
 
   // Serializes all data needed to setup the new replacement channel
   // in the content process into the RedirectToRealChannelArgs struct.
-  void SerializeRedirectData(RedirectToRealChannelArgs& aArgs,
-                             bool aIsCrossProcess, uint32_t aRedirectFlags,
-                             uint32_t aLoadFlags,
-                             dom::ContentParent* aParent) const;
+  void SerializeRedirectData(
+      RedirectToRealChannelArgs& aArgs, bool aIsCrossProcess,
+      uint32_t aRedirectFlags, uint32_t aLoadFlags, dom::ContentParent* aParent,
+      nsTArray<EarlyHintConnectArgs>&& aEarlyHints) const;
 
   uint64_t GetLoadIdentifier() const { return mLoadIdentifier; }
   uint32_t GetLoadType() const { return mLoadStateLoadType; }
@@ -513,6 +514,8 @@
   // switch occurs.
   RefPtr<nsDOMNavigationTiming> mTiming;
 
+  net::EarlyHintsService mEarlyHintsService;
+
   // An optional ObjectUpgradeHandler which can be used to upgrade an <object>
   // or <embed> element to contain a nsFrameLoader, allowing us to switch them
   // into a different process.